pkg/systemd/init.js

```diff
@@ -823,4 +827,490 @@ define([
+        $(".datepicker-dropdown").css("margin-top", $("#timer-dialog").scrollTop());
+        $(".datepicker-dropdown").css("visibility", "visible");
+        $(".datepicker-dropdown .next").show();
+        $(".datepicker-dropdown .prev").show();
+    });
+    // This avoids datepicker incorrect positioning when a click occurs inside it.
+    $("#timer-dialog").on('click', ".datepicker.datepicker-dropdown.dropdown-menu", function() {
+        if (scroll_top_datepicker > scroll_top)
+            $(".datepicker.datepicker-dropdown.dropdown-menu").css("margin-top", scroll_top_datepicker);
+        else
+            $(".datepicker.datepicker-dropdown.dropdown-menu").css("margin-top", scroll_top);
+    });
+    // Calculates the new position when a scroll occurs.
+    $("#timer-dialog").on('mouseenter', ".datepicker.datepicker-dropdown [class*='datepicker-'] thead", function() {
+        scroll_top_datepicker = $("#timer-dialog").scrollTop();
+    });
+
+    $(".form-table-ct").on("click", "[value]", ".btn-group.bootstrap-select.dropdown.form-control", function(ev) {
+        var target = $(this).closest(".btn-group.bootstrap-select.dropdown.form-control");
+        $("span", target).first().text(ev.target.text);
+        $("span", target).first().attr("value", ev.currentTarget.value);
+        switch(target.attr('id')) {
+            case "boot-or-specific-time" : set_boot_or_calendar(Number(ev.currentTarget.value));
+            break;
+            case "drop-time" : set_boot_time_unit(Number(ev.currentTarget.value));
+            break;
+            case "drop-repeat" : repeat_options(Number(ev.currentTarget.value));
+            break;
+        }
+    });
+    // Initialises create timer modal to default options.
+    function timer_init() {
+        $("#command").val("");
+        $("#description").val("");
+        $("#servicename").val("");
+        set_boot_or_calendar(1);
+        $("span", $("#boot-or-specific-time")).first().text("After system boot");
+        $("span", $("#drop-time")).first().text("Seconds");
+        $("span", $("#drop-time")).first().attr("value", "1");
+        repeat_array = [ ];
+        $(".form-control").removeClass("has-error");
+        $(".has-error").hide();
+        timer_unit = {
+            Calendar_or_Boot : "Boot",
+            boot_time_unit :"sec",
+            repeat : "Don't Repeat"
+        };
+    }
+
+    function repeat_options(val) {
+        if ($("#specific-time-error-row").is(":visible")) {
+            $("#specific-time-error-row").hide();
+            $("#hr").removeClass("has-error");
+            $("#min").removeClass("has-error");
+        }
+        switch(val) {
+            case 0 : $("#specific-time-without-repeat").show();
+                $("#repeat-time-option").hide();
+                $("#close_button").hide();
+                $("#hr").val("00");
+                $("#min").val("00");
+                timer_unit.repeat = "Don't Repeat";
```

Comment:
Will these phrases be translated later? If they aren't, they should be translated here: `_("Don't Repeat")`